src/domain/tagset/tagset.service.ts

```diff
@@ -28,7 +35,28 @@ export class TagsetService {
 
     // Check the incoming tags and replace if not null
     tagset.tags = newTags;
-    await tagset.save();
+    await this.tagsetRepository.save(tagset);
+
+    return tagset;
+  }
+
+  async addTag(tagsetID: number, newTag: string): Promise<ITagset> {
+    const tagset = (await this.getTagset(tagsetID)) as Tagset;
+
+    if (!tagset) throw new Error(`Tagset with id(${tagsetID}) not found!`);
+    if (!tagset.tags)
+      throw new Error(`Tagset with id(${tagsetID}) not initialised!`);
+
+    // Check if the tag already exists or not
```

Comment:
Use [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) instead of iterating.

Reply:
You are comparing different types, ignore my comment.